src/fsharp/TypeChecker.fs

```diff
@@ -1079,8 +1079,13 @@ let MakeMemberDataAndMangledNameForMemberVal(g,tcref,isExtrinsic,attrs,optImplSl
              let text = if memberFlags.MemberKind <> MemberKind.Constructor && memberFlags.MemberKind <> MemberKind.ClassConstructor && not memberFlags.IsInstance then text^".Static" else text
              let text = if memberFlags.IsOverrideOrExplicitImpl then text^".Override" else text
              text
+        else if not optIntfSlotTys.IsEmpty then
+            // interface implementation
+            if optIntfSlotTys.Length > 1 then
```

Comment:
Can this condition ever be true?

Reply:
ping @dsyme, I looked a bit at the call-sites, and from my understanding, it will either pass a list with one element, or an empty list. Is that correct?

Reply:
I think it doesn't matter. You are trying to keep the previous mangled name behavior that was there before while adding the new mangled name behavior for multiple generic interfaces. I believe you don't have to worry about that and it's ok to replace the old mangled name behavior entirely. 

The method with that mangled name is private anyway and it only matters when you are actually implementing the interface. If you are using a previously compiled DLL with the old mangled name and you re-implement the interface with the new mangled name behavior, it will have the new name but it will still override on the previous name; therefore, this should be safe to do.

Reply:
I see, you are just trying to keep the same behavior for non-interface types. That makes sense and a conservative way to go about this.